app/lib/three_scale/semantic_form_builder.rb

```diff
@@ -19,10 +19,12 @@ def inline_errors
       @options_inline_erros || super
     end
 
-    def text_field_with_errors( method, opts = {})
-      errors =  @object.errors[method]
+    def text_field_with_errors(method, opts = {})
+      errors = @object.errors[method]
       if errors.present?
         opts = opts.merge(:class => 'error', 'data-errors' => errors, :title => errors)
+      elsif method == :pattern && @object.pattern.in?(['', '/'])
```

Comment:
mmmmm... The problem of this is being called `text_field_with_errors` and being in `app/lib/three_scale/semantic_form_builder.rb`, it seems like it can be called from anywhere in the code, and it expects to just show errors from the model by this name of the method.
So I checked from where it is currently called:
![image](https://user-images.githubusercontent.com/11318903/54941511-4d44fc00-4f2d-11e9-8387-5bb7df3ed050.png)
And thankfully it won't break anything because it is being called only from this same file that you are editing, and you are checking that it is only for the `pattern`, but it still doesn't seem ideal to me 🤔 

I'm thinking what to do about this 😂 I will comment again 😉

Reply:
Those methods are meant to be model agnostic.
`@object` does not necessary have `#pattern` method.

Definitely not the right place. It looks also to me that if there are errors and the pattern is `/` then the tooltip will not show which is not ideal at all.

the form builder gives you accessors to the object by the way:

https://api.rubyonrails.org/v4.2/classes/ActionView/Helpers/FormBuilder.html `[RW] object`

You can just make your conditional in the view:

```ruby
<%= p.text_field_with_errors :pattern, options_for_tipsy_pattern(p.object) %>
```